File: js-pr/56/javascript/code-quality-readability-maintainability/email_validator_14.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
function validateEmail(email) {
    if (!email) return false;
    if (!email.includes('@')) return false;
    let parts = email.split('@');
    if (parts.length !== 2) return false;
    let [local, domain] = parts;
    if (!local || !domain) return false;
    if (!domain.includes('.')) return false;
    let domainParts = domain.split('.');
    if (domainParts.length < 2) return false;
    for (let part of domainParts) {
        if (!part) return false;
    }
    return true;
}
// {/fact}

File: python-pr/52/python/code-quality-readability-maintainability/email_validator_14.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
def validate_email(email):
    if not email: return False
    if '@' not in email: return False
    parts = email.split('@')
    if len(parts) != 2: return False
    local, domain = parts
    if not local or not domain: return False
    if '.' not in domain: return False
    domain_parts = domain.split('.')
    if len(domain_parts) < 2: return False
    for part in domain_parts:
        if not part: return False
    return True
# {/fact}